flow/layers/layer_tree.cc

```diff
@@ -52,9 +55,35 @@ bool LayerTree::Preroll(CompositorContext::ScopedFrame& frame,
       device_pixel_ratio_};
 
   root_layer_->Preroll(&context, frame.root_surface_transformation());
+
+  RasterCache(frame, &context);
+
   return context.surface_needs_readback;
 }
 
+void LayerTree::RasterCache(CompositorContext::ScopedFrame& frame,
+                            PrerollContext* context) {
+  for (unsigned i = 0; i < context->raster_cached_entries.size(); i++) {
+    auto& entry = context->raster_cached_entries[i];
+    if (entry.need_caching) {
+      PrerollContext context = {entry.raster_cache,
+                                frame.gr_context(),
+                                frame.view_embedder(),
+                                entry.mutators_stack,
+                                entry.color_space,
```

Comment:
This value can probably be deduced from the ScopedFrame or gr_context. Check.

Reply:
you are right, thanks, fixed it, done